Cleanup gundeck

fisx opened the pull request:
no semantic changes.  please review commit-by-commit.

neongreen commented:
LGTM, though I don't know enough about https://github.com/wireapp/wire-server/pull/628/commits/ced8e899680dc5ff236297fd70850b5d128a2666 and https://github.com/wireapp/wire-server/pull/628/commits/5a3f73f37da54e09b7e51024e36b2c728339e7f0 to approve.

fisx commented:
> I admit I am no lens expert but given `makeLenses ''Address`, why do we need this?

that generates lenses for the fields of the `Address` record, but i have removed some of those fields and added a `PushToken` field instead that contains exactly the same information.  for old code to still work, i've added the lenses that are disappeared by this change as shortcuts into the now nested type.

not sure i get the question, though?

fisx commented:
> LGTM, though I don't know enough about [ced8e89](https://github.com/wireapp/wire-server/commit/ced8e899680dc5ff236297fd70850b5d128a2666) and [5a3f73f](https://github.com/wireapp/wire-server/commit/5a3f73f37da54e09b7e51024e36b2c728339e7f0) to approve.

[5a3f73f](https://github.com/wireapp/wire-server/commit/5a3f73f37da54e09b7e51024e36b2c728339e7f0) is just changing the structure of a type, without changing its behavior.  you shouldn't need a lot of context to judge this.

[ced8e89](https://github.com/wireapp/wire-server/commit/ced8e899680dc5ff236297fd70850b5d128a2666) removes phantom types that used to have different values, but now don't any more (except in one place in the tests where it either doesn't matter or the flow that doesn't exist in production was tested).

neongreen commented:
> you shouldn't need a lot of context to judge this.

I know it doesn't change the behavior. I don't know if it makes sense semantically.

tiago-loureiro commented:
> that generates lenses for the fields of the `Address` record, but i have removed some of those fields and added a `PushToken` field instead that contains exactly the same information. for old code to still work, i've added the lenses that are disappeared by this change as shortcuts into the now nested type.

> not sure i get the question, though?

Nevermind, just re-read the code 🤦‍♂️ Makes sense :)

ChrisPenner commented on services/gundeck/src/Gundeck/Push/Websocket.hs:
```diff
@@ -98,8 +98,10 @@ logBadCannons (uri, (err, prcs)) = do
-                        show (uri, resource prc, show err))
+            ~~ Log.field "cannon_uri" (show uri)
+            ~~ Log.field "resource_target" (show $ resource prc)
+            ~~ Log.field "http_exception" (show $ show err)  -- (double-show to kill newlines)
```
How does the double show work?

When I try it out it just gets worse:

`> show "test\ntwo\nthree"
"\"test\\ntwo\\nthree\""
`

fisx replied:
worse is better!  :-)

the important thing here is to avoid printing actual newlines to stdout in the middle of a log message.  `\\\\\\\n` is fine.

with some luck there will be a better fix soon: https://gitlab.com/twittner/tinylog/merge_requests/3

ChrisPenner replied:
Ahh, gotcha. Until then perhaps `squashLog = intercalate " | " . lines` would make this look less like it's just a typo

ChrisPenner commented:
The failing ubuntu unit tests are due to old cabal files gunking up the file system; most likely due to the wire-server-ubuntu-cache. I'm rebuilding it using latest develop now and hopefully this will be resolved :D

tiago-loureiro commented:
> The failing ubuntu unit tests are due to old cabal files gunking up the file system; most likely due to the wire-server-ubuntu-cache. I'm rebuilding it using latest develop now and hopefully this will be resolved :D

Yup, looks good now 👍

fisx commented:
@tiago-loureiro wanna approve, then?  :)

fisx commented:
> https://github.com/wireapp/wire-server/pull/628/files/d9038c80c4a273da964500c0cccc7856f439996c#r257470841
> 
> You didn't like that suggestion? :)

I liked it a lot, and followed it: https://github.com/wireapp/wire-server/pull/628/commits/7e58cb919da0069583ea9c2ef9da26793ceaf6a2

I think there is something wrong with how github shows the diff.  Couldn't find the change near the comment.  I'll  merge and double check it'll make it to develop.
